### ISSUE TITLE
Autocomplete broken for certain formatting

### ISSUE THREAD

**Issue by nandorojo** [0no-co/GraphQLSP#190]
If I type the following (using this exact spacing) I get no autocomplete:

```tsx
const mutationDoc = graphql(`mutation MyMutation {
  
}
`)
```

I'm trying to get suggestions for mutation names. However, I just see this:

<img width="620" alt="image" src="https://github.com/0no-co/GraphQLSP/assets/13172299/f3bcff6b-2ef5-4c53-b30d-090562fc5674">

## Workaround

If I add a space before the closing braces, then the formatting does work:

<img width="563" alt="image" src="https://github.com/0no-co/GraphQLSP/assets/13172299/fc2d475e-bcbe-47b7-802f-1a523a0af6e5">

Notice on `line 24`, there is now a space before the closing brace.

It's a bit inconsistent, but I can reproduce it like this:

https://github.com/0no-co/GraphQLSP/assets/13172299/abee5b45-44a9-448e-9741-1ed5380c0db3

Basically, it happens until some formatting changes.

**Comment by nandorojo** [comment]
@JoviDeCroock I'm still getting this issue on 1.2.0, can we reopen?

<img width="965" alt="Screenshot 2024-01-23 at 4 49 04 PM" src="https://github.com/0no-co/GraphQLSP/assets/13172299/6a5ca821-0613-430e-8485-c40b261d5170">

**Comment by JoviDeCroock** [comment]
@nandorojo PR's are welcome 🙌

**Comment by Jarrku** [comment]
Im having similar issues with autocompletes popping up in unexpected locations, and sometimes typescript errors not showing at all. Not sure if I should create a new issue, but I can reproduce it locally with some minor changes in an example:
https://github.com/Jarrku/GraphQLSP/pull/1/files

Basically, it seems to me when nesting fragments that have fragments, something is going wrong with the  typechecking/inline hints for me.

**Comment by JoviDeCroock** [comment]
@Jarrku that's a different issue, this seems to be related to offset compensation which is implemented for tagged templates but not for callExpressions.

Basically when we stub in a fragment we have to re-align the cursor-positioning for subsequent invocations. This would be a similar solution to tackling template-spans

**Comment by kitten** [comment]
I looked into this, and trying to reproduce this, the issue basically seems to happen when the cursor is just one `\n` newline character away from a closing `}` brace, which basically messes up the “GraphQL location.” I’d assume, converting this back, that GraphQLSP makes the autocompletions look at the location of `{` rather than the empty space prior.
